### PR TITLE
Visible panel should have focus after start up #930

### LINKED ISSUE #930: Visible panel should have focus after start up
After HT starts up, the visible panel (i.e. the first panel) should have the focus. I should be able to go to first issue by pressing `Ctrl+down`.

### PR DESCRIPTION
**Pull request by HansNewbie**
Fixes #930

### CODE REVIEW

**Comment by damithc** [comment]
Is it possible to modify tests so that this behavior is confirmed. It will prevent future regressions.

**Comment by HansNewbie** [comment]
Yep, I will do that.

**Comment by HansNewbie** [comment]
May need to change when if #933 is implemented. For now, I will follow `MenuControlTest` in getting `PanelControl`.

**Comment by HansNewbie** [comment]
@damithc I can't make the test unless I change many base classes. I have 2 options to test that the focus is on the first panel upon start up which loads multiple panels:

1. I create many panels, quit the application then launch it again. However, a test can only launch the application once. This option seems to be impossible to work around.
2. I temper the preference test file so that some panels will be loaded on start up. The problem here is that the base class `UITest` - used for all guitests - will delete the the preference test file before launching. Since this will change the base class of all guitests, I am not sure if this is a wise option.

Anyone in the team I can consult about this?

**Comment by HansNewbie** [comment]
@damithc please ignore my previous comment. I talked to Josh about it. Thanks.

File: src/main/java/ui/issuepanel/PanelControl.java [hunk 1]
```diff
@@ -116,6 +117,10 @@ public FilterPanel addPanelAt(int index) {
         return panel;
     }
 
+    public void setFirstPanelAsSelectedPanel() {
```
Comment (dariusf) [inline src/main/java/ui/issuepanel/PanelControl.java]: `panels.selectFirstPanel`?

File: src/test/java/guitests/PanelFocusTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        globalConfig.setPanelInfo(panels);
+        configFileHandler.saveGlobalConfig(globalConfig);
+    }
+
+    @Test
+    public void panelFocusOnFirstPanelAtStartupTest() {
+        PanelControl panelControl = (PanelControl) find("#dummy/dummy_col0").getParent();
+
+        sleep(120000);
```
Comment (dariusf) [inline src/test/java/guitests/PanelFocusTest.java]: The UI tests are run sequentially, so this will add 2 minutes to the entire thing... You can use `UITest. waitUntilNodeAppears` instead with some panel id.

Comment (HansNewbie) [inline src/test/java/guitests/PanelFocusTest.java]: Oops. It was not necessary, I was just debugging something. I will clean it
up.

File: src/test/java/guitests/PanelFocusTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        configFileHandler.saveGlobalConfig(globalConfig);
+    }
+
+    @Test
+    public void panelFocusOnFirstPanelAtStartupTest() {
+        PanelControl panelControl = (PanelControl) find("#dummy/dummy_col0").getParent();
+
+        assertEquals(3, panelControl.getNumberOfPanels());
+        assertEquals(0, (int) panelControl.getCurrentlySelectedPanel().get());
+    }
```
Comment (damithc) [inline src/test/java/guitests/PanelFocusTest.java]: Is one test enough? I thought this is applicable in several places such as opening a new board. Also cover boundary cases.

Comment (HansNewbie) [inline src/test/java/guitests/PanelFocusTest.java]: I did a test for focus on opening a new board in `MenuControlTest` as the test covers opening and closing panel while saving and loading board. Should I move it out?

If I do boundary cases for 0 and 1 panel, I will need a new file for each since I can only launch file once. If that is fine, I will do that.

Comment (HansNewbie) [inline src/test/java/guitests/PanelFocusTest.java]: @lejolly is there any way I can launch HubTurbo multiple times in a single test file? So that I can test with 0 panel, 1 panel and multiple panels on start up.

Comment (damithc) [inline src/test/java/guitests/PanelFocusTest.java]: If it is too expensive, simply add a comment to explain which cases are omitted and why.

Comment (HansNewbie) [inline src/test/java/guitests/PanelFocusTest.java]: @damithc Noted.

**Comment by HansNewbie** [comment]
@dariusf anything else I need to change?

File: src/test/java/guitests/PanelFocusTest.java (hunk 3 above)
Comment (lejolly) [inline src/test/java/guitests/PanelFocusTest.java]: @HansNewbie nope, so far we've only been able to get one JavaFX application to start per test class.

**Comment by dariusf** [comment]
No, unless you're planning to add more tests?

**Comment by HansNewbie** [comment]
@dariusf no more tests.

**Comment by HansNewbie** [comment]
Sorry, I did the pull request merge wrongly so this is not closed automatically. The commit that should have done that is ca91dbb84d91d0e735423cc560476643f08b544c.

**Comment by dariusf** [comment]
The latest changes are in.
